assignment7.txt

```diff
@@ -0,0 +1,16 @@
+1. For an array {1, 2, 3, 4}
+	a. ++*p will get the variable located at *p then increment it. if *p is at index 0, new array would be {2, 2, 3, 4} with *p at index 0
+	b. *p++ will increment p (move the index forward once), then get the value of it. If *p is at index 0, then the array will be {1,2,3,4} with *p at index 1
+	c. *++p is equivalent to *p++ as the prefix ++ and * evaluate right to left.
+2. No, it depends on the type of operator.
+3. Pointers are used for functions that need to actually change their arguments or for returning an array of values
```

Comment:
-Pointers are more efficient in handling arrays and data tables 
-They can be used to return multiple values from a function via function arguments 
-Pointers permit references to functions 
-Pointers allow C to support dynamic memory management 
-Pointers reduce length and complexity of programs